docs/02.deploying/11.restore/11.restore.md

```diff
@@ -7,23 +7,84 @@ slug: /deploying/restore
+https://github.com/neuvector/neuvector/security/advisories/GHSA-h773-7gf7-9m2x
+
+:::
 
 :::note
-NeuVector does not support partial restoration of objects (e.g. network rules only) nor timestamped restoration (e.g. restore from date/time snapshots). Please use [automation scripts](/automation/automation#exportimport-configuration-file) to regularly backup configuration files and manage timestamps.
+NeuVector does not support partial restores (for example, restoring only network rules) or point-in-time restores. Use automation scripts to regularly back up configuration files and manage timestamped backups. For more information, see [Export and import configuration files](/automation/automation#exportimport-configuration-file).
 :::
 
 :::warning important
-The backup configuration files should not be edited in any way. Any changes to these from their exported state could result in restoration errors and an unpredictable result. 
+Do not modify backup configuration files. Editing a backup file after export can cause restore failures or result in an unpredictable system state.
 :::
 
 :::info caution
-Backup configuration files should be used to restore a NeuVector state on the same cluster from which they were exported. Applying a backup configuration file from a different cluster could result in unpredictable results.
+Use backup configuration files only to restore NeuVector on the same cluster from which they were exported. Restoring a backup to a different cluster may result in unpredictable behavior.
+:::
+
+## Encryption key management
+
+The `neuvector-store-secret` contains the **Key Encryption Key (KEK)**. NeuVector uses the KEK to generate **Data Encryption Keys (DEKs)** that encrypt sensitive configuration data.
+
+Key behavior:
+
+- You must maintain and back up the `neuvector-store-secret`.
+- If the secret is missing or the key value does not meet length requirements, NeuVector automatically creates or updates it.
+- When NeuVector creates or updates the secret, it generates an alert reminding you to back it up.
+- If the KEK changes or is lost, previously encrypted data cannot be decrypted.
+
+During rolling upgrades or restores:
+
+1. Data encrypted with the legacy hard-coded key is re-encrypted using a DEK derived from the KEK.
+2. Data encrypted using a DEK derived from the KEK can be decrypted only by NeuVector using the same KEK.
+
+![Rolling upgrade event](rolling-upgrade-event.png)
+
+If re-encryption occurs, NeuVector records the action and generates an event.
+
+## Sensitive data protected by encryption
+
+NeuVector encrypts sensitive data stored under the following key-value paths:
+
+- `object/config/server/ldap1`
+- `object/config/server/oidc1`
+- `object/config/server/saml1`
+- `object/config/system`
+- `object/config/registry`
+- `object/config/federation/membership`
+- `object/config/federation/clusters/...`
+- `object/cert/...`
+
+![Neuvector store secret ss](neuvector-store-secret.png)
```

Comment:
What does `ss` stand for in `Neuvector store secret ss` ?

Reply:
Aah you are right, sorry. ss in images usually stands for screenshot.

Reply:
Fixed it